include/solvers/time_solver.h

```diff
@@ -132,12 +134,29 @@ class TimeSolver : public ReferenceCountedObject<TimeSolver>
    */
   virtual void retrieve_timestep();
 
+  /**
+   * A method to integrate the system::QoI functionals
+   */
+  virtual void integrate_qoi_timestep();
+
   /**
    * A method to integrate the adjoint sensitivity w.r.t a given parameter
    * vector. int_{tstep_start}^{tstep_end} dQ/dp dt = int_{tstep_start}^{tstep_end} (\partialQ / \partial p) - ( \partial R (u,z) / \partial p ) dt
    */
   virtual void integrate_adjoint_sensitivity(const QoISet & qois, const ParameterVector & parameter_vector, SensitivityData & sensitivities);
 
+  /**
+   * A method to compute the adjoint refinement error estimate at the current timestep.
+   * int_{tstep_start}^{tstep_end} R(u^h,z) dt
+   * The user provides an initialized ARefEE object.
+   * Fills in an ErrorVector that contains the weighted sum of errors from all the QoIs and can be used to guide AMR.
+   * As an option, the user can also supply a QoI_time_instant vector where non-null pointers imply an instantaneous
+   * QoI evaluation at the time held by the pointer.
+   * Also fills in a map that links QoI indices to spatially integrated error estimates for the current timestep.
+   * ONLY SUPPORTED for Backward Euler.
```

Comment:
Let's put "ONLY CURRENTLY SUPPORTED", and in the subclasses put an
```
if (theta != 1.0)
  libmesh_not_implemented();
```
to ensure it?  People should read documentation, and should read warnings, but they don't, and what people *do* read is the final message saying why their program stopped.

Reply:
Done.